packages/bluesnap-direct-integration/src/bluesnap-direct-credit-card-payment-strategy.ts

```diff
@@ -0,0 +1,91 @@
+import { WithCreditCardPaymentInitializeOptions } from '@bigcommerce/checkout-sdk/credit-card-integration';
+import {
+    guard,
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    OrderFinalizationNotRequiredError,
+    OrderRequestBody,
+    PaymentArgumentInvalidError,
+    PaymentInitializeOptions,
+    PaymentIntegrationService,
+    PaymentStrategy,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import BlueSnapDirectHostedForm from './bluesnap-direct-hosted-form';
+import isHostedCardFieldOptionsMap from './is-hosted-card-field-options-map';
+
+export default class BlueSnapDirectCreditCardPaymentStrategy implements PaymentStrategy {
+    private _paymentFieldsToken?: string;
+
+    constructor(
+        private _paymentIntegrationService: PaymentIntegrationService,
+        private _blueSnapDirectHostedForm: BlueSnapDirectHostedForm,
+    ) {}
+
+    async initialize(
+        options: PaymentInitializeOptions & WithCreditCardPaymentInitializeOptions,
+    ): Promise<void> {
+        const { methodId, gatewayId, creditCard } = options;
+
+        if (!gatewayId || !creditCard || !isHostedCardFieldOptionsMap(creditCard.form.fields)) {
+            throw new InvalidArgumentError();
+        }
+
+        const {
+            config: { testMode },
+            clientToken,
+        } = (
+            await this._paymentIntegrationService.loadPaymentMethod(gatewayId, {
+                params: { method: methodId },
+            })
+        ).getPaymentMethodOrThrow(methodId, gatewayId);
```

Comment:
```suggestion
        const {
            config: { testMode },
            clientToken,
        } = 
            await this._paymentIntegrationService.loadPaymentMethod(gatewayId, {
                params: { method: methodId },
            })
        .getPaymentMethodOrThrow(methodId, gatewayId);
```

Reply:
I need to await the promise first otherwise, I get:
`Property 'getPaymentMethodOrThrow' does not exist on type 'Promise<PaymentIntegrationSelectors>'.`

Reply:
Then in that case maybe separate them in two lines?